R/workhorse.R

```diff
@@ -27,57 +28,79 @@ build_request <- function(endpoint = NULL,
 ## right now not setting endpoint, if you want to include parameters, must be in a named list
 ## may introduce the :parameter notation in endpoint later :)
 
-set_query <- function(x){
-  if (length(x$params) == 0L) return(x)
+set_query <- function(x) {
+  if (length(x$params) == 0L)
+    return(x)
   if (x$method != "GET") {
     if (grepl("\\?", x$endpoint)) {
       x$query <- sub(".*\\?", "", x$endpoint)
       return(x)
     }
     return(x)
   }
-  if (!all(has_names(x$params))){
+  if (!all(has_names(x$params))) {
     spf("All parameters must be named.")
   }
   x$query <- x$params
   x$params <- NULL
   x
 }
 
-set_body <- function(x){
-  if (length(x$params) == 0L) return(x)
+set_body <- function(x) {
+  if (length(x$params) == 0L)
+    return(x)
 
   x$body <- x$params
   x
 }
 
 ## not setting headers yet
 
-set_url <- function(x){
-
-  if (grepl("^http", x$endpoint)){
+set_url <- function(x) {
+  if (grepl("^http", x$endpoint)) {
```

Comment:
This smells like something that will go away once this is designed to align perfectly against `httr::modify_url()`.